dds_cli/gui_poc/docs/poc_dds_auth.md

```diff
@@ -0,0 +1,94 @@
+# POC on Auth commands with GUI
+
+## DDS CLI 
+
+### Current setup
+Currently no return of the API respone from the middle layer is implemented. Want to make the response availible for the GUI. 
+
+![Image](./DDS_current_implementation.jpeg)
+
+Want to either:
+- Move the logging to the click application and make the cli functions return the API response, or
+- Keep the logging in the cli functions but make the fuction return the API response
+
+### Example of current implementation
+
+```python
+# -- dds user info -- #
+@user_group_command.command(name="info")
+# Options
+# Flags
+@click.pass_obj
+def get_info_user(click_ctx):
+    """Display information connected to your own DDS account.
+
+    Usable by all user roles.
+
+    \b
+    The following information should be displayed:
+    - Username
+    - Role
+    - Name
+    - Primary email
+    - Associated emails (not useful yet)
+    """
+    try:
+        with dds_cli.account_manager.AccountManager(
+            no_prompt=click_ctx.get("NO_PROMPT", False),
+            token_path=click_ctx.get("TOKEN_PATH"),
+        ) as get_info:
+            get_info.get_user_info()
+    except (
+        dds_cli.exceptions.APIError,
+        dds_cli.exceptions.AuthenticationError,
+        dds_cli.exceptions.DDSCLIException,
+        dds_cli.exceptions.ApiResponseError,
+        dds_cli.exceptions.ApiRequestError,
+    ) as err:
+        LOG.error(err)
+        sys.exit(1)
+```
+
+```python
+def get_user_info(self):
+        """Get a users info."""
+        response, _ = dds_cli.utils.perform_request(
+            dds_cli.DDSEndpoint.DISPLAY_USER_INFO,
+            headers=self.token,
+            method="get",
+            error_message="Failed to get user information",
+            timeout=dds_cli.DDSEndpoint.TIMEOUT,
+        )
+
+        for field in response.get("info", []):
+            if isinstance(response["info"][field], str):
+                response["info"][field] = rich.markup.escape(response["info"][field])
+
+        info = response.get("info")
+        if info:
+            LOG.info(
+                "Username:          %s \n"
+                "Role:              %s \n"
+                "Name:              %s \n"
+                "Primary Email:     %s \n"
+                "Associated Emails: %s \n",
+                info["username"],
+                info["role"],
+                info["name"],
+                info["email_primary"],
+                ", ".join(str(x) for x in info["emails_all"]),
+            )
+```
+
+### Comments 
+
+- For some cli functions it's enough to just add a return statement in the method
+- The authentication would need to be either reformated or adding a parallel authentication for the gui --> could be a better solution to rewrite the authenticatin flow all together 
```

Comment:
Yes we don't want the parallel authentication, we want to reduce the amount of duplicate code as much as possible, so I agree that the better solution is to refactor the authentication flow